### DISCUSSION TITLE
Alternative to faker.address.streetSuffix()?

### DISCUSSION THREAD

**Discussion by filetvignon** [faker-js/faker#1844] [Q&A]
The docs say the function faker.address.streetSuffix() is deprecated (https://fakerjs.dev/api/address.html#streetsuffix)

But I can't find information on what the alternative is

**Comment by ST-DDT** [comment]
There is no alternative. Could you describe what you need the suffix for?

**Reply by filetvignon** [reply]
I actually needed streetSuffix, but it has the same issue.

I need to generate addresses in different formats, say, 1075 Wayland Street vs 1075 Wayland St.

Is there any easy way to do this?

**Reply by ST-DDT** [reply]
faker.address.street() / faker.location.street() will use the suffixes to generate these variants for you (with different name parts though, assuming that the name parts are irrelevant for testing).

Alternatively, you can access the locale data directly (works, but not considered official api):

````ts
faker.helpers.arrayElement( faker.definitions.address.street_suffix );
````

**Reply by filetvignon** [reply]
Thanks! This works.

Are suffixes only single words? I mean, can I use address.street() and then consider that last word generated to be the suffix?

The issue I have with address.street() is that it never uses abbreviations like "St" or "Ave", which is quite common in real databases, so I'm doing something to randomize that

**Reply by matthewmayer** [reply]
In the default en locale they are all currently single words

You could probably just have a fixed list of replacements like Street to St, Avenue to Ave etc.

**Reply by ST-DDT** [reply]
> The issue I have with address.street() is that it never uses abbreviations like "St" or "Ave", which is quite common in real databases, so I'm doing something to randomize that

Should we extend our list of suffixes then?

**Reply by filetvignon** [reply]
I found a comprehensive list online of suffixes, their abbreviations and common misspellings and I'm using that to randomly generate data with all possibilities
